crates/gossip/src/peer_meta_store.rs

```diff
@@ -0,0 +1,174 @@
+        )
+        .await
+    }
+
+    /// Get the "new ops" bookmark for the given agent.
+    ///
+    /// This is a value provided by the agent to indicate the last op they have
+    /// sent us. We can use this to request an incremental update from them.
+    pub(crate) async fn new_ops_bookmark(
+        &self,
+        peer: Url,
+    ) -> K2Result<Option<Timestamp>> {
+        self.get(peer, "gossip:new_ops_bookmark").await
+    }
+
+    /// Set the "new ops" bookmark for the given agent.
+    pub(crate) async fn set_new_ops_bookmark(
+        &self,
+        peer: Url,
+        timestamp: Timestamp,
+    ) -> K2Result<()> {
+        self.put(
+            peer,
+            "gossip:new_ops_bookmark",
+            timestamp,
+            // Try to keep this value around for 24 hours. If we check in once a day, it seems
+            // reasonable to remember where we were syncing from. After that, fall back to catchup
+            // logic.
+            Some(Timestamp::now() + Duration::from_secs(24 * 60 * 60)),
+        )
+        .await
+    }
+
+    async fn get<T: serde::de::DeserializeOwned>(
+        &self,
+        peer: Url,
+        name: &str,
+    ) -> Result<Option<T>, K2Error> {
+        self.inner
+            .get(self.space.clone(), peer, name.to_string())
+            .await?
+            .map(|v| {
+                serde_json::from_slice::<T>(&v).map_err(|e| {
+                    K2Error::other_src(
+                        format!(
+                            "failed to deserialize peer meta value for {}",
+                            name
+                        ),
+                        e,
+                    )
+                })
+            })
+            .transpose()
+    }
+
+    async fn put<T: serde::Serialize>(
+        &self,
+        peer: Url,
+        name: &str,
+        value: T,
+        expiry: Option<Timestamp>,
+    ) -> K2Result<()> {
+        let value = serde_json::to_vec(&value).map_err(|e| {
+            K2Error::other_src("failed to serialize peer meta value", e)
+        })?;
+
+        self.inner
+            .put(
+                self.space.clone(),
+                peer,
+                name.to_string(),
+                value.into(),
+                expiry,
+            )
+            .await?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use kitsune2_core::default_test_builder;
+    use kitsune2_core::factories::MemPeerMetaStoreFactory;
+    use std::sync::Arc;
+
+    async fn test_store() -> K2PeerMetaStore {
+        let space = SpaceId::from(bytes::Bytes::from_static(b"test"));
+        let builder =
+            Arc::new(default_test_builder().with_default_config().unwrap());
+        let inner = MemPeerMetaStoreFactory::create()
+            .create(builder)
+            .await
+            .unwrap();
+        K2PeerMetaStore::new(inner, space)
+    }
+
+    #[tokio::test]
+    async fn create() {
+        test_store().await;
```

Comment:
This seems to be sufficiently tested implicitly by the other tests.

Reply:
Yep, written while I was getting the module set up, not needed

Reply:
Do you want to keep it anyway?

Reply:
No, I think removing it is the right call, it's gone